### ISSUE TITLE
RayCast not colliding if origin is inside an Area

### ISSUE THREAD

**Issue by SilvanaP** [godotengine/godot#43496]
**Godot version:**
3.1

**OS/device including version:**
Windows

**Issue description:**
I am desparetely tring to cast a ray from inside an area and get the collision with that area's collision shape. I found out that this is working as expected when the ray originates from outside the area, but when the ray is located inside the area, no collision is reported. To me, it seems that backfaces are not considered when raycasting. Is this a bug? A feature? Or do I miss a switch to enable intersection with backfaces in order to raycast from within an area/collision shape?

**Steps to reproduce:**
Make an area with a collision shape (and optionally, for visualisation, a mesh). Cast a ray from outside into the area: collision point detected, works like a charm. Cast a ray from inside that area to the outside: no collision detected.

**Comment by mrjustaguy** [comment]
1. MRP
2. Try the latest stable to see if it still happens there

**Comment by SilvanaP** [comment]
@mrjustaguy Yes, it's still present in 3.2.3 and it actually makes sense to check collisions only with frontfaces and not with backfaces in most setups. 

I worked around this sow by creating a special collision mesh in Blender with inversed normals (so they are pointing inward my mesh, not outward). I use this mesh as collision shape for my area and voila, collisions are detected when casting a ray from inside. 

It would still be nice to have a switch to enable intersection with backfaces in order to raycast from within an area/collision shape without making custom collision shapes with weird normals. Maybe someone can move this thread to suggestions? (From what I found on the Web, Unity also introduced such a switch at some point it seems.)
